### ISSUE TITLE
Consider adding cop for detecting protected method definitions in singleton classes

### ISSUE THREAD

**Issue by preetpalS** [rubocop-hq/rubocop#5577]
In ruby, protected methods only make sense as instance methods of a class. Protected methods defined on a singleton class do not differ behaviourally from private methods (ignoring inheritance) on a singleton class as singleton classes cannot be instantiated (the only major difference is that these methods are classified differently).

```ruby
# Sample class to highlight issue.
class Sample
  class << self
    # Protected method on singleton class (might be interpreted as a "protected
    # class method" (behaviourally equivalent to a private class method))
    protected def say(x)
      puts x
    end
  end
end

s = Sample.new
sample_singleton_class = (class << s; self; end)
puts sample_singleton_class.protected_methods.include?(:say) # true
puts sample_singleton_class.private_methods.include?(:say) # false

# It is impossible to create instance of Sample singleton class
begin
  _sample_singleton_class_instance = sample_singleton_class.new
rescue TypeError => e
  puts e.message
end

# Protected class methods on a Singleton class can still be called but
# they most likely are a symptom of misunderstanding.
Sample.send(:say, 'Hi')
```

Also see https://stackoverflow.com/questions/48529615/why-does-the-instance-method-protected-class-method-not-exist-for-the-module.

--------

## Expected behavior

Rubocop should detect (and warn on) protected method definitions in singleton classes.

## Actual behavior

Rubocop issues no warnings for protected method definitions in singleton classes.

## Steps to reproduce the problem

Sample terminal session.

```
$ rubocop sample.rb  
Inspecting 1 file
Warning: /home is a symlink, which is not allowed.
.

1 file inspected, no offenses detected

$ cat sample.rb 

# Sample class to highlight issue.
class Sample
  class << self
    # Protected method on singleton class (might be interpreted as a "protected
    # class method" (behaviourally equivalent to a private class method))
    protected def say(x)
      puts x
    end
  end
end

s = Sample.new
sample_singleton_class = (class << s; self; end)
puts sample_singleton_class.protected_methods.include?(:say) # true
puts sample_singleton_class.private_methods.include?(:say) # false

# It is impossible to create instance of Sample singleton class
begin
  _sample_singleton_class_instance = sample_singleton_class.new
rescue TypeError => e
  puts e.message
end

# Protected class methods on a Singleton class can still be called but
# they most likely are a symptom of misunderstanding.
Sample.send(:say, 'Hi')

$ ruby sample.rb 
true
false
can't create instance of singleton class
Hi
```

## RuboCop version

```
$ rubocop -V
0.52.1 (using Parser 2.4.0.2, running on ruby 2.5.0 x86_64-freebsd11.1)
```

**Comment by bbatsov** [comment]
There hasn't been much activity on this ticket and our Core Team is spread too thin on so many tasks, so I'll just close it for the sake of having a cleaner lists of tasks to focus on. We'd gladly review a PR, but it's unlikely that'd we're going to tackle this ourselves in the foreseeable future.

**Comment by preetpalS** [comment]
I personally do not think that this cop would be that useful since I have never seen a protected method definition on a singleton class outside of this example (which mainly came as a result of exploring the language). If someone has actually encountered this before, they should make a case for adding this feature.
